lib/http/chainable.rb

```diff
@@ -74,15 +74,15 @@ def patch(uri, options = {}) # rubocop:disable Style/OptionHash
     # @param uri
     # @option options [Hash]
     def request(verb, uri, options = {}) # rubocop:disable Style/OptionHash
-      branch(options).request verb, uri
+      branch(default_options.merge(options)).request verb, uri
     end
 
     # Prepare an HTTP request with the given verb
     # @param verb
     # @param uri
     # @option options [Hash]
     def build_request(verb, uri, options = {}) # rubocop:disable Style/OptionHash
-      branch(options).build_request verb, uri
+      branch(default_options.merge(options)).build_request verb, uri
```

Comment:
In both cases this and above, it will be more effiecient to pass options to delegated method instead:

``` ruby
# @param (see Client#build_request)
def build_request(*args)
  branch(default_options).build_request(*args)
end
```

This way you will avoid double-merging with no good. :D